src/resolvers/resolver.cr

```diff
@@ -87,6 +87,7 @@ module Shards
     abstract def read_spec(version : Version) : String?
     abstract def install_sources(version : Version, install_dir : String)
     abstract def report_version(version : Version) : String
+    abstract def update_local_cache
```

Comment:
I'm not sure if this makes sense as an abstract method when the concept doesn't even apply to all resolvers (`Path`, `Crystal` don't have a local cache). Instead, we could avoid empty implementations and guard the call to this method by `responds_to?`. Alternatively, we could add an empty implementation here and override it only in ancestors that have a meaningful implementation.

Reply:
Hmm. True.

Changed to an empty def in https://github.com/crystal-lang/shards/pull/539/commits/7f45434b3e5f20dea105b01af7a086d122bcc3ac

(I think that's the more discoverable variant of the two, but no strong opinion here,
just let me know if you'd prefer `responds_to?` instead)